Added documentation for swift (#10)
DataDog/ciapp-alpha-docs: docs/tests/swift.md

---
layout: page
title:  Swift
categories: swift test
permalink: /tests/swift
parent: Test Instrumentation
---

# Installation

## Using SPM

Add 'dd-sdk-swift-testing' package to your project. It is located at https://github.com/DataDog/dd-sdk-swift-testing.
Then link your test targets with the library `DatadogSDKTesting` from the package. 

> If you are doing UITesting, your app running the UItests should also be linked with this library

Reviewer: ```suggestion
If you are doing UITesting, your app running the UItests should also be linked with this library.
```

## Binary linking

> Download the project at  [repository](https://github.com/DataDog/dd-sdk-swift-testing)

Reviewer: ```suggestion
Download the project at [dd-sdk-swift-testing](https://github.com/DataDog/dd-sdk-swift-testing)
```

> In terminal go to the project folder and run `make release`, the resulting framework `DatadogSDKTesting.xcframework` can be found at : `./build/xcframework`

Reviewer: ```suggestion
In a terminal go to the project folder and run `make release`, the resulting framework `DatadogSDKTesting.xcframework` can be found at `./build/xcframework`
```

> Then you can copy and link your test targets with the XCFramework. If you are doing UITesting, your app running the UItests should also be linked with this framework

Reviewer: ```suggestion
Then you can copy and link your test targets with the XCFramework. If you are doing UITesting, your app running the UItests should also be linked with this framework.
```

## Enabling

### Datadog Confguration

> To enable testing instrumentation you must dd the following environment variables to your test target. For UITests environment variables only need to be set in the test target, since the framework automatically injects these values to the application.

Reviewer: ```suggestion
To enable testing instrumentation you must add the following environment variables to your test target. For UITests environment variables only need to be set in the test target, since the framework automatically injects these values to the application.
```

| Key                      | Value                       |
|--------------------------|-----------------------------|
| `DD_TEST_RUNNER`              | `true`              |
| `DATADOG_CLIENT_TOKEN`              | `<your current Datadog Client Token>`              |

You may want to set other environment variables also:

| Key                      | Value                       |
|--------------------------|-----------------------------|

> | `DD_ENV`              | `<The environment you want to report>`              |

Reviewer: ```suggestion
| `DD_ENV`              | `<The environment you want to report>`              |
```

| `GIT_REPOSITORY_URL`                          | `$(GIT_REPOSITORY_URL)`                          |
| `BITRISE_GIT_COMMIT`                | `$(BITRISE_GIT_COMMIT)`                |
| `BITRISE_SOURCE_DIR`               | `$(BITRISE_SOURCE_DIR)`               |
| `BITRISE_APP_TITLE`             | `$(BITRISE_APP_TITLE)`             |
| `BITRISE_BUILD_SLUG`             | `$(BITRISE_BUILD_SLUG)`             |
| `BITRISE_BUILD_NUMBER`                 | `$(BITRISE_BUILD_NUMBER)`                 |
| `BITRISE_BUILD_URL`                      | `$(BITRISE_BUILD_URL)`                      |
| `BITRISE_APP_URL`                  | `$(BITRISE_APP_URL)`                  |
| `BITRISE_GIT_BRANCH`                 | `$(BITRISE_GIT_BRANCH)`                 |
| `BITRISEIO_GIT_BRANCH_DEST` | `$(BITRISEIO_GIT_BRANCH_DEST)` |
| `BITRISE_GIT_TAG` | `$(BITRISE_GIT_TAG)` |
| `GIT_CLONE_COMMIT_HASH` | `$(GIT_CLONE_COMMIT_HASH)` |

## Running tests 

After installation, you can run your tests as you normally do, for example using the `xcodebuild test` command. 
Tests, network requests and application logs will be instrumented automatically.

## Extra configuration

### Disabling Auto Instrumentation

> The framework automatically tries to capture the maximum information, but for some situations or tests it can be counter-productive. You can disable some of the autoinstrumentation for all the tests, by setting the following environment variables. 

Reviewer: ```suggestion
The framework automatically tries to capture the most information possible, but for some situations or tests it can be counter-productive. You can disable some of the autoinstrumentation for all the tests by setting the following environment variables:
```

> >Boolean variables can use any of: "1","0","true","false", "YES", "NO"

Reviewer: ```suggestion
>Boolean variables can use any of: "1", "0", "true", "false", "YES", "NO"
```

> >String List variables accepts a list of elements separated by "," or ";"

Reviewer: ```suggestion
>String list variables accept a list of elements separated by "," or ";"
```

{% highlight txt %}
DD_DISABLE_NETWORK_INSTRUMENTATION # Disables all network instrumentation (Boolean)
DD_DISABLE_STDOUT_INSTRUMENTATION # Disables all stdout instrumentation (Boolean)
DD_DISABLE_STDERR_INSTRUMENTATION # Disables all stderr instrumentation (Boolean)
{% endhighlight %}

### Network Auto Instrumentation

For Network autoinstrumentation there are other settings that you can configure

{% highlight txt %}
DD_DISABLE_HEADERS_INJECTION # Disables all injection of tracing headers (Boolean)
DD_INSTRUMENTATION_EXTRA_HEADERS # Specific extra headers that you want the tool to log (String List)

> DD_EXCLUDED_URLS # Urls that you dont want to log or inject headers into(String List)

Reviewer: ```suggestion
DD_EXCLUDED_URLS # Urls that you don't want to log or inject headers into (String List)
```
